ArduCopter/mode_autorotate.cpp

```diff
@@ -82,27 +81,68 @@ void ModeAutorotate::run()
         set_mode(copter.prev_control_mode, ModeReason::AUTOROTATION_BAILOUT);
     }
 
+    float pilot_roll, pilot_pitch, pilot_yaw_rate;
+    if (g2.arot._param_guided == 1){
+        pilot_roll = guided_roll;				
+    } else {
+        // Operator is in control of roll and yaw.  Controls act as if in stabilise flight mode.  Pitch 
+        // is controlled by speed-height controller.	
+        get_pilot_desired_lean_angles(pilot_roll, pilot_pitch, copter.aparm.angle_max, copter.aparm.angle_max);						
+    }
+		
+    float target_roll;
+
+    // Grab inertial velocity
+    const Vector3f& vel = inertial_nav.get_velocity_neu_cms();
+
+    // rotate roll, pitch input from north facing to vehicle's perspective
+    float roll_vel =  vel.y * ahrs.cos_yaw() - vel.x * ahrs.sin_yaw(); // body roll vel
+    float pitch_vel = vel.y * ahrs.sin_yaw() + vel.x * ahrs.cos_yaw(); // body pitch vel
+
+    // gain scheduling for yaw
+    float pitch_vel2 = MIN(fabsf(pitch_vel), 2000);
```

Comment:
I would really like to try the weather vaneing library for doing the coordinated turns.  Would reduce this code and minimise duplication.

Reply:
yes, this was to do some testing of the control logic. Having the nose aligned onto velocity vector  would be a great improvement both for avoiding sideslip during glide and for precise skid aligning during landing.